x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/AuthorizedProjectsSupplier.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.security.authz;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.search.crossproject.TargetProjects;
+
+/**
+ * A supplier of authorized projects for the current user. This includes the origin project and all linked projects the user has access to.
+ * If we are not in a cross-project search context, the supplier returns {@link TargetProjects#NOT_CROSS_PROJECT}.
+ */
+public interface AuthorizedProjectsSupplier {
+    void getAuthorizedProjects(ActionListener<TargetProjects> listener);
+
+    boolean recordResolvedIndexExpressions();
```

Comment:
It does not feel like this belongs here. It mixes project resolving with index resolving.
Seems to me its purpose is to signal if CPS is enabled or not?